src/xcode/ENA/ENA/Source/Services/Risk/Calculation/RiskCalculation.swift

```diff
@@ -75,33 +74,34 @@ enum RiskCalculation {
 			appDelegate?.lastRiskCalculation.append("summary: \(String(describing: summary?.description))\n")
 		}
 
+		//
 		// Precondition 1 - Exposure Notifications must be turned on
-		guard preconditions.isGood else {
-			// This overrides all other levels
-			return .success(.inactive)
-		}
+		let isInactive = !preconditions.isGood
 
 		// Precondition 2 - If tracing is active less than 1 day, risk is .unknownInitial
-		if activeTracing.inHours < minTracingActiveHours, riskLevel < .unknownInitial {
-			riskLevel = .unknownInitial
-		}
+		let isTracingActiveLess1Day = activeTracing.inHours < minTracingActiveHours
 
 		// Precondition 3 - Risk is unknownInitial if summary is not present
-		if summary == nil, riskLevel < .unknownInitial {
-			riskLevel = .unknownInitial
-		}
+		let isNoSummary = summary == nil
 
-		if
-			!providerConfiguration.exposureDetectionIsValid(lastExposureDetectionDate: dateLastExposureDetection ?? .distantPast),
-			riskLevel < .unknownOutdated {
-			// The last exposure detection is not valid since it occurred too far in the past
-			riskLevel = .unknownOutdated
-		}
+		let isUnknownInitial = isTracingActiveLess1Day || isNoSummary
 
-		guard let summary = summary else {
-			return .success(riskLevel)
+		let isUnknownOutdated = !providerConfiguration.exposureDetectionIsValid(lastExposureDetectionDate: dateLastExposureDetection ?? .distantPast)
+
+		var riskLevels: [RiskLevel] = [.low]
+
+		// returns RiskLevel with higher priority
+		var riskLevel: RiskLevel {
+			riskLevels.max() ?? .inactive
 		}
```

Comment:
This approach is cool, I've never actually used such a computed variable before - never thought of it 👍

Reply:
I also have never used it before, just tried to write it and surprisingly it works 😅